python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pandas as pd
+from bigdl.nano.utils.log4Error import invalidInputError
+import logging
+
+
+def quality_check_timeseries_dataframe(df,
+                                       dt_col):
+    '''
+    detect the low-quality data and provide suggestion (e.g. call .impute or .resample).
+
+    :param df: a pandas dataframe for your raw time series data.
+    :param dt_col: a str indicates the col name of datetime
+            column in the input data frame, the dt_col must be sorted
+            from past to latest respectively for each id.
+    '''
+    invalidInputError(dt_col in df.columns, f"dt_col {dt_col} can not be found in df.")
+    invalidInputError(pd.isna(df[dt_col]).sum() == 0, "There is N/A in datetime col")
+    flag = True
+    # 1. timestamp check
+    flag = flag and _timestamp_type_check(df[dt_col])
+
+    # 2. irregular interval check
+    flag = flag and _time_interval_check(df[dt_col])
+
+    # 3. missing value check
+    flag = flag and _missing_value_check(df, dt_col)
+
+    # 4. pattern check and noise check
+    # TODO:
+
+    return flag
+
+
+def _timestamp_type_check(df_column):
```

Comment:
add comment here to explain what this checker is used for

Reply:
done.